checkov/common/checks/base_check.py

```diff
@@ -84,7 +84,8 @@ def run(
 
     @multi_signature()
     @abstractmethod
-    def scan_entity_conf(self, conf: dict[str, Any], entity_type: str) -> CheckResult | tuple[CheckResult, dict[str, Any]]:
+    def scan_entity_conf(self, conf: dict[str, Any], entity_type: str) -> \
+            CheckResult | tuple[CheckResult, dict[str, Any] | str | list[str | dict[str, Any]]]:
```

Comment:
@arielkru This return type is just impossible. No idea what to expect here TBH. This is a bad sign